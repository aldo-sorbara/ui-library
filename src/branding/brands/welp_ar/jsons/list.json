{
  "padding": "0 0 0 1rem",
  "color": "#0082FF"
}
