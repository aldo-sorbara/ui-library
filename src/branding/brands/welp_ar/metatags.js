export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
