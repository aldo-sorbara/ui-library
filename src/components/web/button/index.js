export { Button, ButtonContainer } from './Button';
