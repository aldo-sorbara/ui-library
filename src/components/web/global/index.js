export {
  Container,
  Wrapper,
  Headline,
  Description,
  Legales,
  Label,
  LabelLarge,
  SecondHeadline,
  ContainerDesktop,
  Desktop,
  Onboarding,
  Terms,
  H1,
  SectionDescription,
  UrlLink,
  Timer,
  DesktopText,
  Bind,
  Logo,
  LogoMobile,
  TextPromotion
} from './Global';
