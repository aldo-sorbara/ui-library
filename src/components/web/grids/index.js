export { Grid, Container as GridContainer } from './Grids';
