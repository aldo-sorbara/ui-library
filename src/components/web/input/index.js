export { default, InputLabel, InputDescription } from './Input';
