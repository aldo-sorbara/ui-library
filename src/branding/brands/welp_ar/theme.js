import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
