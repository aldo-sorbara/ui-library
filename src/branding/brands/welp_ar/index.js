import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
