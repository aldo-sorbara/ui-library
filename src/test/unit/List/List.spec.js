import React from 'react';
import { List, ListItem } from '../../../components/web/list';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import welp_ar from '../../../branding/brands/welp_ar';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

describe('List component', () => {
  it('should render List snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <List className='test-list' data='list'>
          <ListItem>Test 1</ListItem>
          <ListItem>Test 2</ListItem>
          <ListItem>Test 3</ListItem>
        </List>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render list with default props', () => {
    const wrapper = mountWithTheme(
      <List className='test-list' data='default-list'>
        <ListItem>Test 1</ListItem>
        <ListItem>Test 2</ListItem>
        <ListItem>Test 3</ListItem>
      </List>,
      welp_ar.theme
    );

    expect(wrapper).toHaveStyleRule('margin', '0');
  });
  it('should render listItem with default props', () => {
    const wrapper = mountWithTheme(<ListItem>Test 1</ListItem>, welp_ar.theme);

    expect(wrapper).toHaveStyleRule('margin-bottom', '.5rem');
    expect(wrapper).toHaveStyleRule('padding', '0 0 0 1rem');
    expect(wrapper).toHaveStyleRule('margin-bottom', '0', {
      modifier: ':last-child'
    });
    expect(wrapper).toHaveStyleRule('top', '.25rem', {
      modifier: ':before'
    });
  });
});
