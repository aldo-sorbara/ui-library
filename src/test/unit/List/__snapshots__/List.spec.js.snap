// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should render List snapshot 1`] = `
.c0 {
  margin: 0;
}

.c1 {
  margin-bottom: .5rem;
  font-size: .875rem;
  font-weight: 400;
  line-height: 1.25;
  padding: 0 0 0 1rem;
}

.c1:last-child {
  margin-bottom: 0;
}

.c1:before {
  background-color: #0082FF;
  border-color: #0082FF;
  top: .25rem;
}

<ul
  className="test-list c0"
  data="list"
>
  <li
    className="c1"
  >
    Test 1
  </li>
  <li
    className="c1"
  >
    Test 2
  </li>
  <li
    className="c1"
  >
    Test 3
  </li>
</ul>
`;
